pkg/test/utils/kubernetes.go

```diff
@@ -789,19 +789,25 @@ func WaitForCRDs(dClient discovery.DiscoveryInterface, crds []runtime.Object) er
 
 	for _, crdObj := range crds {
 		if !MatchesKind(crdObj, crdKind) {
-			continue
+			return fmt.Errorf("the following passed object does not match %v: %v", crdKind, crdObj)
 		}
 
-		crd, ok := crdObj.(*apiextensions.CustomResourceDefinition)
-		if !ok {
-			continue
+		switch crd := crdObj.(type) {
+		case *apiextensions.CustomResourceDefinition:
+			waitingFor = append(waitingFor, schema.GroupVersionKind{
+				Group:   crd.Spec.Group,
+				Version: crd.Spec.Version,
+				Kind:    crd.Spec.Names.Kind,
+			})
+		case *unstructured.Unstructured:
+			waitingFor = append(waitingFor, schema.GroupVersionKind{
```

Comment:
What happens if the unstructured object is not a CRD?

Reply:
Then it will [be rejected](https://github.com/kudobuilder/kudo/pull/1284/files#diff-d78eb5efc1e23fc1c46369c7b1377002R792).